api/schemas/source.json

```diff
@@ -0,0 +1,59 @@
+{
+    "type": "object",
+    "description": "Describes a Source: an abstract representation of a piece of media as defined in <https://specs.amwa.tv/ms-04/releases/v1.0.0/docs/2.2._Explanation_-_Source.html>\n\nSources may be elemental (and represented directly by a Flow), or may represent a collection of other Sources, e.g. a Source collecting video and audio together.",
+    "title": "Source",
+    "required": [
+        "id"
+    ],
+    "properties": {
+        "id": {
+            "description": "Source identifier",
+            "type": "string",
+            "pattern": "^[0-9a-f]{8}-[0-9a-f]{4}-[1-5][0-9a-f]{3}-[89ab][0-9a-f]{3}-[0-9a-f]{12}$"
+        },
+        "label": {
+            "description": "Freeform string label for the Source",
+            "type": "string"
+        },
+        "description": {
+            "description": "Freeform text describing the Source",
+            "type": "string"
+        },
+        "created_by": {
+            "description": "A string identifier for the entity that created the Source. Implementations SHOULD set suitable default values for `created_by` based on the principal accessing the system, and MAY permit clients to edit the value, subject to suitable permissions-based limitations.",
+            "type": "string"
+        },
+        "modified_by": {
+            "description": "A string identifier for the entity that modified the Source metadata most recently. Implementations SHOULD set suitable default values for `modified_by` based on the principal accessing the system, and MAY permit clients to edit the value, subject to suitable permissions-based limitations.",
+            "type": "string"
+        },
+        "created": {
+            "description": "The date-time the Source was created in a given context, e.g. in a store",
+            "type": "string",
+            "format": "date-time"
+        },
+        "updated": {
+            "description": "The date-time the Source metadata was last updated in a given context, e.g. in a store",
+            "type": "string",
+            "format": "date-time"
+        },
+        "tags": {
+            "$ref": "tags.json"
+        },
+        "contributes_to": {
+            "type": "array",
+            "description": "Sources which collect this Source to form part of them, i.e parents of this Source. Only present if `include_hierarchy` is set.",
+            "items": {
+                "type": "string",
+                "pattern": "^[0-9a-f]{8}-[0-9a-f]{4}-[1-5][0-9a-f]{3}-[89ab][0-9a-f]{3}-[0-9a-f]{12}$"
+            }
+        },
+        "contains": {
+            "type": "array",
+            "description": "Sources which this Source collects together, i.e. children of this Source. Only present if `include_hierarchy` is set.",
```

Comment:
It might be sensible to add a "SHOULD" recommendation for implementations to populate the reverse `contributes_to` fields based on `contains` rather than leaving it up to the client to populate. This could very easily lead to partial/inconsistent data.